examples/widgets/slider_demo.py

```diff
@@ -5,63 +5,70 @@
 
 Using the slider widget to control visual properties of your plot.
 
-In this example, a slider is used to choose the frequency of a sine
-wave. You can control many continuously-varying properties of your plot in
-this way.
+In this example, sliders are used to control the frequency and amplitude of
+a sine wave. You can control many continuously-varying properties of your plot
+in this way.
+
+For a more detailed example of value snapping see
+:doc:`/gallery/widgets/slider_snap_demo`.
+
+For an example of using a `matplotlib.widgets.RangeSlider` to define a range
+of values see :doc:`/gallery/widgets/range_slider`.
 """
 import numpy as np
 import matplotlib.pyplot as plt
-from matplotlib.widgets import Slider, Button, RadioButtons
+from matplotlib.widgets import Slider, Button
+
+
+def fxn(t, amp, freq):
+    return amp * np.sin(2 * np.pi * freq * t)
 
-fig, ax = plt.subplots()
-plt.subplots_adjust(left=0.25, bottom=0.25)
 t = np.arange(0.0, 1.0, 0.001)
+
+# Define initial parameters
 a0 = 5
 f0 = 3
-delta_f = 5.0
-s = a0 * np.sin(2 * np.pi * f0 * t)
-l, = plt.plot(t, s, lw=2)
-ax.margins(x=0)
+
+# Create the figure and the `~.Line2D` that we will manipulate
+fig, ax = plt.subplots()
+line, = plt.plot(t, fxn(t, a0, f0), lw=2)
 
 axcolor = 'lightgoldenrodyellow'
+ax.margins(x=0)
+
+# adjust the main plot to make room for the sliders
+plt.subplots_adjust(left=0.25, bottom=0.25)
+
+# Make a horizontal slider to control the frequency.
+# This slider will snap to discrete values as defind by ``valstep``.
 axfreq = plt.axes([0.25, 0.1, 0.65, 0.03], facecolor=axcolor)
-axamp = plt.axes([0.25, 0.15, 0.65, 0.03], facecolor=axcolor)
+freq_slider = Slider(axfreq, 'Freq', 0.1, 30.0, valinit=f0, valstep=5.0)
```

Comment:
Maybe
```suggestion
freq_slider = Slider(axfreq, 'Freqency', 0.1, 30.0, valinit=f0, valstep=5.0)
```

Reply:
:+1: seems reasonable to me. same to amplitude